cmd/common/image/inspect.go

```diff
@@ -18,85 +18,154 @@ import (
+		}
+
+		runtimeGlobalParams := runtime.GlobalParamDescs().ToParams()
+		runtimeParams := runtime.ParamDescs().ToParams()
+		ociParams := apihelpers.ToParamDescs(ocihandler.OciHandler.InstanceParams()).ToParams()
+
+		// Add operator global flags
+		opGlobalParams := make(map[string]*params.Params)
+		for _, op := range operators.GetDataOperators() {
+			opGlobalParams[op.Name()] = apihelpers.ToParamDescs(op.GlobalParams()).ToParams()
+		}
+
+		var info *api.GadgetInfo
+		err = runtime.Init(runtimeGlobalParams)
+		if err != nil {
+			return fmt.Errorf("initializing runtime: %w", err)
+		}
+		defer runtime.Close()
+
+		// set global operator flags from the config file
+		for o, p := range opGlobalParams {
+			err = common.SetFlagsForParams(cmd, p, config.OperatorKey+"."+o)
 			if err != nil {
-				return fmt.Errorf("inspecting image: %w", err)
+				return fmt.Errorf("setting operator %s flags: %w", o, err)
 			}
+		}
 
-			switch outputMode {
-			case utils.OutputModeJSON:
-				bytes, err := json.Marshal(image)
-				if err != nil {
-					return fmt.Errorf("marshalling image to JSON: %w", err)
-				}
-				fmt.Fprint(cmd.OutOrStdout(), string(bytes))
-			case utils.OutputModeJSONPretty:
-				bytes, err := json.MarshalIndent(image, "", "  ")
-				if err != nil {
-					return fmt.Errorf("marshalling image to JSON: %w", err)
-				}
-				fmt.Fprint(cmd.OutOrStdout(), string(bytes))
-			case utils.OutputModeColumns:
-				isTerm := term.IsTerminal(int(os.Stdout.Fd()))
-
-				cols := columns.MustCreateColumns[oci.GadgetImageDesc]()
-				if isTerm {
-					cols.MustSetExtractor("digest", func(i *oci.GadgetImageDesc) any {
-						if i.Digest == "" {
-							return ""
-						}
-						// Return the shortened digest and remove the sha256: prefix
-						return strings.TrimPrefix(i.Digest, "sha256:")[:12]
-					})
-					now := time.Now()
-					cols.MustSetExtractor("created", func(i *oci.GadgetImageDesc) any {
-						if t, err := time.Parse(time.RFC3339, i.Created); err == nil {
-							return fmt.Sprintf("%s ago", strings.ToLower(units.HumanDuration(now.Sub(t))))
-						}
-						return ""
-					})
-				}
+		ops := make([]operators.DataOperator, 0)
+		for _, op := range operators.GetDataOperators() {
+			// Initialize operator
+			err := op.Init(opGlobalParams[op.Name()])
+			if err != nil {
+				continue
+			}
+			ops = append(ops, op)
+		}
+
+		gadgetCtx := gadgetcontext.New(
+			context.Background(),
+			image.String(),
+			gadgetcontext.WithDataOperators(ops...),
+			gadgetcontext.WithUseInstance(false),
+			gadgetcontext.IncludeExtraInfo(true),
+		)
 
-				formatter := textcolumns.NewFormatter(cols.GetColumnMap(), textcolumns.WithShouldTruncate(isTerm))
-				formatter.WriteTable(cmd.OutOrStdout(), []*oci.GadgetImageDesc{image})
-			default:
-				return fmt.Errorf("invalid output mode %q, valid values are: %s", outputMode, strings.Join(outputModes, ", "))
+		paramValueMap := make(map[string]string)
+		ociParams.CopyToMap(paramValueMap, "operator.oci.")
```

Comment:
Could you use the parameter pull=never?
https://github.com/inspektor-gadget/inspektor-gadget/blob/v0.38.0/pkg/operators/oci-handler/oci.go#L138-L149


The current documentation of the inspect command says:
```
inspect     Inspect the local gadget image
```
But for me, the inspect command downloads things from ghcr.io. I think it should not download anything.